### ISSUE TITLE
Excluding files from `Rails` department in config doesn't disable all the `Rails` cops

### ISSUE THREAD

**Issue by ojab** [rubocop/rubocop#11148]
Given `.rubocop.yml`
```yaml 
require:
  - rubocop-rails

AllCops:
  NewCops: enable
  
Rails:
  Exclude:
    - ./lib/1.rb
```
and `./lib/1.rb`
```ruby
# frozen_string_literal: true

puts 123
exit 1
```

--------

## Expected behavior

No offenses reported.

## Actual behavior

```
$ rubocop --debug --cache=false
For /tmp/tst: configuration from /tmp/tst/.rubocop.yml
configuration from .../rubocop-rails-2.17.2/config/default.yml
configuration from .../rubocop-rails-2.17.2/config/default.yml
Default configuration from.../rubocop-1.38.0/config/default.yml
Inspecting 1 file
Scanning /tmp/tst/lib/1.rb
C

Offenses:

lib/1.rb:4:1: C: Rails/Exit: Do not use exit in Rails applications.
exit 1
^^^^

1 file inspected, 1 offense detected
Finished in 0.25235605498892255 seconds
```
Note, without excluding the file there is also `Rails/Output` offense reported, so at least some cops are disabled.

## Steps to reproduce the problem

above

## RuboCop version
```
1.38.0 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.2) [x86_64-linux]
  - rubocop-rails 2.17.2
```

**Comment by splattael** [comment]
Note that it works for other Rails cops like `Rails/NegateInclude`.

Consider this file `lib/1.rb`:

```ruby
# frozen_string_literal: true

x = ![].include?(x)
puts x

exit 1
```

Without department exclude I see 3 offenses (as expected):

```
$ rubocop -C false
Inspecting 1 file
C

Offenses:

lib/1.rb:3:5: C: [Correctable] Rails/NegateInclude: Use .exclude? and remove the negation part.
x = ![].include?(x)
    ^^^^^^^^^^^^^^^
lib/1.rb:4:1: C: [Correctable] Rails/Output: Do not write to stdout. Use Rails's logger if you want to log.
puts x
^^^^
lib/1.rb:6:1: C: Rails/Exit: Do not use exit in Rails applications.
exit 1
^^^^

1 file inspected, 3 offenses detected, 2 offenses autocorrectable
```

Exclude `Rails` department again via

```yaml

Rails:
  Exclude:
    - ./lib/1.rb
```

shows one remaining Rails offense:

```
 rubocop -C false
Inspecting 1 file
C

Offenses:

lib/1.rb:6:1: C: Rails/Exit: Do not use exit in Rails applications.
exit 1
^^^^

1 file inspected, 1 offense detected
```

Is this because of the default `Include`? :thinking: 

```
$ rubocop --show-cops Rails/Exit
Rails/Exit:
  Description: Favor `fail`, `break`, `return`, etc. over `exit` in application or library
    code outside of Rake files to avoid exits during unit testing or running in production.
  Enabled: true
  VersionAdded: '0.41'
  Include:
  - app/**/*.rb
  - config/**/*.rb
  - lib/**/*.rb
  Exclude:
  - "/home/peter/devel/testbed-rubocop/lib/**/*.rake"
```

**Comment by corsonknowles** [comment]
This has been bothering me -- I am glad to have found an open issue about it.

**Comment by Earlopain** [comment]
I think this is actually a duplicate of https://github.com/rubocop/rubocop/issues/9983 😅. I tried fixing this somewhat recently but the config code is rather complex and I didn't really manage.
